### ISSUE TITLE
Proposed (eventually) breaking object syntax change

### ISSUE THREAD

**Issue by nesquena** [nesquena/rabl#216]
Issue #183 started an interested discussion between me and @databyte and @blakink. What if we changed the syntax to group the content into object/collection blocks. For instance, take this example from today:

``` ruby
collection @posts => :posts

attributes :id, :name, :body, :created_at

node :is_read do |p|
  p.read_by? @user
end

child :author do |p|
  attributes :name, :username, :email
end
```

Now consider this syntax change where the rules are logically grouped into the collection with a block:

``` ruby
collection :posts => @posts do |p|
  attributes :id, :name, :body, :created_at

  node :is_read do |p|
    p.read_by? @user
  end

  child :author do |p|
    attributes :name, :username, :email
  end
end
```

I find the second way more consistent and intuitive IMO. I think we could potentially deprecate in 0.6.0 and remove it in 0.7.0. This is a beta and I am not afraid to be willing to switch syntax if we can agree that it's a step in the right direction. 

It also makes adding root nodes easier or even multiple objects:

``` ruby
collection :posts => @posts do |p|
  # ...
end

node :total_pages do
   @posts.total_pages
end

node :current_page do
  @posts.current_page
end
```

as well as accessing the object in a consistent way:

``` ruby
collection :posts => @posts do |p|
  if p.author.admin?
     # ... group based on the post in the collection
  end
end
```
## Summary

In short, here's what we want to do:
### Change object syntax into a block

``` ruby
object @post do |p|
  attributes :id, :name, :body, :created_at
end
```
### Change collection syntax into a block

``` ruby
collection :posts => @posts do |p|
  attributes :id, :name, :body, :created_at

  node :is_read do |p|
    p.read_by? @user
  end

  child :author do |p|
    attributes :name, :username, :email
  end
end
```
### Reverse hash syntax from

``` ruby
collection @posts => :posts
```

to

``` ruby
collection :posts => @posts
```
### Stricter handling of collection and object.

If the developer says 'object' then assume object, if they say collection then assume collection. We no longer will detect this ourselves.
